ribs/emitters/rankers.py

```diff
@@ -333,25 +343,29 @@ def rank(self, emitter, archive, rng, solution_batch, objective_batch,
 }
 
 
-def get_ranker(name):
+def _get_ranker(klass):
     """Returns a ranker class based on its name.
 
-    ``name`` may be the full name of a ranker, e.g. "ImprovementRanker" or
-    "RandomDirectionRanker". Alternatively, it can be the abbreviated name
-    for a ranker -- the supported abbreviations are:
-
-    * ``imp``: :class:`ImprovementRanker`
-    * ``2imp``: :class:`TwoStageImprovementRanker`
-    * ``rd``: :class:`RandomDirectionRanker`
-    * ``2rd``: :class:`TwoStageRandomDirectionRanker`
-    * ``obj``: :class:`ObjectiveRanker`
-    * ``2obj``: :class:`TwoStageObjectiveRanker`
+    ``klass`` can be a reference to the class of the ranker, the full name of
+    a ranker, e.g. "ImprovementRanker", or the abbreviated name for a ranker
+    such as "imp".
 
     Args:
-        name (str): Full or abbreviated name of the ranker.
+        klass (Callable or str): This parameter may either be a callable (e.g.
+            a class or a lambda function) that takes in no parameters and
+            returns an instance of :class:`RankerBase`, or it may be a full or
+            abbreviated ranker name.
     Returns:
         The corresponding ranker class.
     """
-    if name in _NAME_TO_RANKER_MAP:
-        return _NAME_TO_RANKER_MAP[name]
-    raise ValueError(f"Could not find ranker with name {name}")
+    if isinstance(klass, str):
+        klass = _NAME_TO_RANKER_MAP.get(klass)
```

Comment:
I think you should still check if klass is in the map. Right now you are relying on the fact that `.get` will return None by default and `callable(None)` is not True.